### PR TITLE
Xenomorph/Alien Rework 2023: Part 1

### PR DESCRIPTION
**Pull request by IndieanaJones**
## About The Pull Request

Alternative to #75277

Kept you waiting, huh?

This PR is the first part of a Xenomorph rework which seeks to make the big lugs more balanced and up to date with /tg/'s current design. This mainly involves curtailing xenomorph's infamous hardstuns into more interactive forms of combat, while also giving some buffs to the xenomorph's more unique abilities in order to keep them threatening.

Part 1 will focus on simple number changes and some simple mechanic changes. In the future, changes will be made to endgame involving xenomorphs, along with changes to other facets of Xenomorphs.

Highly based off of #55937.

Changes:

- Xenomorph disarm has been completely reworked. While a disarm will attempt to, well, disarm, a human opponent should they be holding something, it will no longer immediately hardstun targets when they aren't. Instead, the xenomorph will shove the target several tiles back and inflict 35 stamina damage. If the target slams into a wall, this will also come with the added effect of knocking them down. If a human is incapacitated, however, right click will slam them into the ground, which paralyzes them for a lengthy 5 seconds (which is ultimately half the time xenos could stun you for before), allowing for safe transport back to the nest as long as you keep them close.

- Humans can now shove xenomorphs. Due to being the superior predator, however, you can't knock down xenomorphs from shoving. You can slow them for a little bit akin to humans though.

- Neurotoxin no longer is a hardstun. Instead, it deals 50 stamina damage on contact. It is still resisted by BIO armor.

**HUNTER:**
- Speed reduced from -1 to -0.3.
- Pounce speed is twice as fast as before (1 to 2)
- Hardstun time on pounce reduced from 10 seconds to 5 seconds.

Hunters being insanely fast has been a major balance-ruining factor of xenomorphs for many years now. These buggers could practically ambush anyone, hardstun them immediately, and then leave before anyone could do anything. Now, with their speed nerfed and in combination with the xeno shove changes, hunters will need to spend more time to down a target. Their pounce was practically useless, so its been sped up in order to make it more practical to use.

**SENTINEL**
- Speed reduced from 0 to 0.2
- Cloak alpha reduced from 0.75 to 0.25 (you're more hidden now)

Sentinels receive a large nerf in regards to their spit, but their before useless cloaking ability has been greatly improved upon as compensation. They now serve better as defenders and ranged ambushers.

**XENOMORPH DRONE**
- No changes

As in the original PR, drones are perfeclty balanced in my eyes, so no changes were required.

**XENOMORPH PRAETORIAN**
- Speed increased from 1 to 0.5
- No changes

Praetorians get affected by the nerfs of the other xeno abilities, but now they're a bit faster in order to close the gap to use their abilities.

**XENOMORPH QUEEN**
- Speed increased from 3 to 2
- Health increased from 400 to 500
- Damage increased from 20 to 50

Xenomorph queens have been sped up and made more tanky and lethal in close-range combat. Fighting this beast up-close should be a death sentence to almost anything else in the game. Speed increases will help her re-position and close the gap on potential prey.

**OTHER CHANGES**
- Fixed a bug where simplemobs didn't actually use xenomorph's damage values when they were attacked by them.

## Why It's Good For The Game

Xenomorphs are old, and haven't been updated for quite a long time. This has left them as sources of a bunch of hardstuns which made counterplay from a modern spaceman extremely difficult. With these changes, fighting xenomorphs is more interactive and should end up being more enjoyable for both crew and xenos. Buffs were also given out to incentivize usage of xenomorph's unique abilities as opposed to the standard disarm spam which was most effective for them until now.

## Changelog
:cl:
balance: Xenos have been rebalanced, removing their hardstuns on their disarm and neurotoxin, along with a slew of other changes. Xenos have received buffs to their more unique abilities in return.
fix: Fixed simplemobs ignoring xenomorph's melee damage values when being attacked by them.
/:cl:

### CODE REVIEW

**Comment by MTandi** [comment]
Can some of this be applied to NPC xenomorphs? It would be cool if they were more dangerous and use some abilities, like carps.

**Comment by Pickle-Coding** [comment]
I thought xenomorphs were supposed to be scary.

**Comment by DaydreamIQ** [comment]
This is a step in the right direction for sure. I know there's gonna be those people who complain that Xenos won't be a threat anymore without their guaranteed 10 second hardstun, but in my experience people will usually just end the round rather than risking fighting a threat that can solo 5 people at once if none of them have a ranged weapon or hulk.

**Comment by Dorsisdwarf** [comment]
> 



> I thought xenomorphs were supposed to be scary.

xenos can be scary in ways other than "barely visible black smear rockets up to you from offscreen at mach 10, instantly hardstuns you and drags you off into the distance while chainstunning and killing you."

**Comment by Helg2** [comment]
> I thought xenomorphs were supposed to be scary.

And all the scariness ends when someone brings an energy gun.

**Comment by Swiftkor** [comment]
Perhaps some thought could be put into discouraging xenos from spacing their nests? It makes it more difficult for the crew to fight (requiring players to put on a hardsuit which slows them down) and has zero downsides for the xenos, who can just disengage into space where most players cannot follow. Xenos should probably keep their space protection, but incentivizing them to keep their victims alive before the larva burst out would address this.

**Comment by IndieanaJones** [comment]
> Perhaps some thought could be put into discouraging xenos from spacing their nests? It makes it more difficult for the crew to fight (requiring players to put on a hardsuit which slows them down) and has zero downsides for the xenos, who can just disengage into space where most players cannot follow. Xenos should probably keep their space protection, but incentivizing them to keep their victims alive before the larva burst out would address this.

This is something I intend to address in a future PR.

File: code/modules/mob/living/carbon/alien/adult/caste/hunter.dm [hunk 1]
```diff
@@ -75,7 +78,7 @@
 					blocked = TRUE
 			if(!blocked)
 				L.visible_message(span_danger("[src] pounces on [L]!"), span_userdanger("[src] pounces on you!"))
-				L.Paralyze(100)
+				L.Paralyze(5 SECONDS)
```
Comment (Iamgoofball) [inline code/modules/mob/living/carbon/alien/adult/caste/hunter.dm:81]: remove the paralyze or this is a meaningless change

**Comment by castawaynont** [comment]
> I thought xenomorphs were supposed to be scary.

**Comment by itseasytosee** [comment]
BRILLIANT MORE STAMINA DAMAGE REPLACING STUNS. I get your intention but its just so damn gamey. Where is the ART where is the FICTION. RNG is more engaging than pure clicking skill.

**Comment by carlarctg** [comment]
you underestimate how much Xenos are impacted by laser weapons and other burn damage. i think that could do with an adjustment on this pr as they are very very deadly

**Comment by KittyNoodle** [comment]
> I thought xenomorphs were supposed to be scary.

Xenos have never felt scary, just has always felt like conflict-less bullshit. There is no thrill of the chase because they move at meth level speeds at all times. There is no interesting fights because they have instant stun unarmed attacks, something the crew could never compete with in the modern game. There is no confrontation because their speed promotes hit and run over anything else.

There are actual examples of currently in game very hard to beat mobs that are scary, heretic worm and statue come to mind. Both of these mobs are scary in their own way, the heretic worm is a melee juggernaut that can break through walls, your only move is to run away or hide, this is both fun and terrifying. The statue on the other hand is a fast moving mob with a catch, it can only move when you can’t see it, this creates a tense environment as you must try to get as far away from it without breaking eye contact, while also getting far enough away to survive its blinding spell.

**Comment by Googles-Hands** [comment]
Mostly alright changes but I don't think you should be able to shove xenos, removing instant hardstuns is one thing but turning it into shovematches is silly.

**Comment by IndieanaJones** [comment]
> Mostly alright changes but I don't think you should be able to shove xenos, removing instant hardstuns is one thing but turning it into shovematches is silly.

You can only shove xenos to apply that small slowdown that humans also get when you shove them. Xenomorphs can't get knocked down if they're shoved into a wall. It's most for a, "I'm unarmed and running for my life" usage where you might be able to slow a xeno down a little bit if you can time your shove.

**Comment by Ekaterina-von-Russland** [comment]
Yes! A sorely needed rework, it looks great!

**Comment by Loafin34** [comment]
could we possibly get a situation with xenos where certain gasses effect them? it'd be fun to see xenos run out of fuel because of a BZ leak and risk endangering future hosts (if they have to be alive) by spacing to get rid of the harmful BZ

**Comment by nukularpower** [comment]
Speaking as someone that views stamcrit as an abomination (compared to old stun gameplay)... Meh.  Would be cool if neurotoxin did some tox damage too if it doesnt already.

**Comment by Helg2** [comment]
> Speaking as someone that views stamcrit as an abomination (compared to old stun gameplay)... Meh. Would be cool if neurotoxin did some tox damage too if it doesnt already.

it deals 5 damage

**Comment by ATH1909** [comment]
Not entirely a fan of sentinels being slower than humans now (if I'm reading this right). I get that you're supposed to use the gamer spit to slow a target down to catch up to them, but god this is going to make fighting hulks and catching people you've yeeted a pain in the ass.

How many shots of neurospit do you get from a full tank? Maybe the cost of neurospit should be halved to reflect the fact that it's only around half as effective (vs. unarmored targets) now.

**Comment by IndieanaJones** [comment]
I will resolve the borg stun issue but I will not change the sentinel's speed.

**Comment by MrrFiish** [comment]
Heya I love this BUT if possible could we also get it in here that either A) you play as your own chestburster or B) it keeps the soul in your head because currently from what I've heard and experienced one of the biggest complaints is that its a guaranteed RR no matter what, so putting you in the chestburster or keeping the soul intact would be lovely I think.

**Comment by Bluedino1025** [comment]
one idea for a future part of this is that instead of gibbing you chestbursting only disembowels you, that way its actually possible to save people when xenos show up.

**Comment by IndieanaJones** [comment]
> The disarm doesn't stun borgs anymore, and you removed the paralysis from neurospit. Are non-hunters kind of just fucked against borgs now?

I'm a dolt, silicons aren't humans (duh). I didn't change anything about the cyborg xenomorph disarm interaction, so it works the same as it did before. You can't use spit to disable cyborgs, but shove still works.

File: code/modules/mob/living/carbon/alien/adult/caste/hunter.dm (hunk 1 above)
Comment (IndieanaJones) [reply code/modules/mob/living/carbon/alien/adult/caste/hunter.dm:81]: Pounce is very hard to land and gets blocked by shields. Its one of those things I'm comfortable with still having a hardstun, and the reason I shortened it was to make it harder to keep people stunned unless you're being more attentive to them.

**Comment by Kendrickorium55** [comment]
about time these guys got reworked, excited to see these changes in game

**Comment by ExcessiveUseOfCobblestone** [comment]
will merge on thaw

**Pull request merged**